### PR TITLE
Add HTTP response code for any HTTP related request

### PR DESCRIPTION
**Pull request by AuraWind**
HTTP response codes are dropped when a successful HTTP request is made through the ExHal.Client.  Update the tuple to return a %ResponseHeader{} for any HTTP related requests

### CODE REVIEW

File: lib/exhal/client.ex [hunk 1]
```diff
@@ -81,17 +81,17 @@ defmodule ExHal.Client do
     case http_resp do
       {:error, err} -> {:error, %ExHal.Error{reason: err.reason} }
 
-      {:ok, resp} -> extract_doc(client, resp)
+      {:ok, resp} -> parse_response(client, resp)
     end
   end
 
-  defp extract_doc(client, resp) do
+  defp parse_response(client, resp) do
```
Comment (pezra) [inline lib/exhal/client.ex]: `parse` -> `interpret` maybe?

File: lib/exhal/response_header.ex [hunk 2]
```diff
@@ -0,0 +1,3 @@
+defmodule ExHal.ResponseHeader do
```
Comment (pezra) [inline lib/exhal/response_header.ex]: Can we get a module doc describing the responsibilities of this module?

Comment (AuraWind) [inline lib/exhal/response_header.ex]: sure can

**Comment by AuraWind** [comment]
README has been updated with new ResponseHeader in the response tuples

**Pull request merged**